### PR TITLE
Issue 404: Install TOC extension by default

### PR DESCRIPTION
**Pull request by vkumra-broad**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: docker/jupyter/scripts/extension/install_TOC.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+#!/usr/bin/env bash
```
Comment (rtitle) [inline docker/jupyter/scripts/extension/install_TOC.sh]: I think this script needs more :)

File: src/main/resources/jupyter/init-actions.sh [hunk 2]
```diff
@@ -247,6 +247,9 @@ if [[ "${ROLE}" == 'Master' ]]; then
       done
     fi
 
+
+    retry 3 docker exec -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_SCRIPTS}/extension/install_TOC.sh
```
Comment (rtitle) [inline src/main/resources/jupyter/init-actions.sh]: Probably should remove the `-u root` -- I think we install extensions as `jupyter-user`.

**Comment by vkumra-broad** [comment]
I am not sure if I like approach, I would rather have this done via a user script than installing the extension by default. Thoughts?

File: docker/jupyter/scripts/extension/install_TOC.sh (hunk 1 above)
Comment (vkumra-broad) [reply docker/jupyter/scripts/extension/install_TOC.sh]: this is what happens when you work during the plenary :)

**Comment by rtitle** [comment]
I'd also lean towards a user script approach for more customizability than doing it by default. Could loop in @sharmatime too.

**Comment by sharmatime** [comment]
Lets review this tomorrow (Thursday) Vikram

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInstallSpec.scala [hunk 3]
```diff
@@ -17,7 +17,7 @@ class NotebookInstallSpec extends ClusterFixtureSpec {
               |bx.bitset.sys.copyright""".stripMargin
 
           notebookPage.executeCell("1+1") shouldBe Some("2")
-          notebookPage.executeCell(getPythonVersion) shouldBe Some("3.6.7")
+          notebookPage.executeCell(getPythonVersion) shouldBe Some("3.6.8")
```
Comment (vkumra-broad) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInstallSpec.scala]: There was a python release recently. This change reflects that.

Comment (vkumra-broad) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInstallSpec.scala]: https://www.python.org/downloads/release/python-368/

**Comment by vkumra-broad** [comment]
jenkins retest

**Comment by vkumra-broad** [comment]
jenkins retest

**Comment by MatthewBemis** [comment]
what is the reason for doing this by default instead of via user script like was discussed above? i really don't think that we should be fielding requests for what's installed by default. i get that this extension is fairly trivial and useful but i think in general, if somebody wants an extension installed by default while using terra then it should be terras responsibility for installing that extension. the customizability is there for terra to do so, and AoU and other apps might not want these things

**Comment by vkumra-broad** [comment]
We went with by default installation since these are commonly used extensions. AoU also uses these extensions for there "getting started notebook", it's just that they install it themselves.

**Pull request merged**